src/qibolab/instruments/zhinst/executor.py

```diff
@@ -35,64 +28,51 @@
 }
 """Translating to Zurich ExecutionParameters."""
 ACQUISITION_TYPE = {
-    AcquisitionType.INTEGRATION: lo.AcquisitionType.INTEGRATION,
-    AcquisitionType.RAW: lo.AcquisitionType.RAW,
-    AcquisitionType.DISCRIMINATION: lo.AcquisitionType.DISCRIMINATION,
+    AcquisitionType.INTEGRATION: laboneq.AcquisitionType.INTEGRATION,
+    AcquisitionType.RAW: laboneq.AcquisitionType.RAW,
+    AcquisitionType.DISCRIMINATION: laboneq.AcquisitionType.DISCRIMINATION,
 }
 
 AVERAGING_MODE = {
-    AveragingMode.CYCLIC: lo.AveragingMode.CYCLIC,
-    AveragingMode.SINGLESHOT: lo.AveragingMode.SINGLE_SHOT,
+    AveragingMode.CYCLIC: laboneq.AveragingMode.CYCLIC,
+    AveragingMode.SINGLESHOT: laboneq.AveragingMode.SINGLE_SHOT,
 }
 
 
-@dataclass
-class ZhPort(Port):
-    name: tuple[str, str]
-    offset: float = 0.0
-    power_range: int = 0
-
-
-@dataclass
-class SubSequence:
-    """A subsequence is a slice (in time) of a sequence that contains at most
-    one measurement per qubit.
-
-    When the driver is asked to execute a sequence, it will first split
-    it into sub-sequences. This is needed so that we can create a
-    separate laboneq section for each measurement (multiple measurements
-    per section are not allowed). When splitting a sequence, it is
-    assumed that 1. a measurement operation can be parallel (in time) to
-    another measurement operation (i.e. measuring multiple qubits
-    simultaneously), but other channels (e.g. drive) do not contain any
-    pulses parallel to measurements, 2. ith measurement on some channel
-    is in the same subsequence as the ith measurement (if any) on
-    another measurement channel, 3. all measurements in one subsequence
-    happen at the same time.
-    """
-
-    measurements: list[tuple[str, ZhPulse]]
-    control_sequence: dict[str, list[ZhPulse]]
+def _acquisition_handle(seq_idx: int, pulse_idx: int, acquisition_name: str) -> str:
+    return f"sequence_{seq_idx}_{acquisition_name}_{pulse_idx}"
 
 
 class Zurich(Controller):
     """Driver for a collection of ZI instruments that are automatically
     synchronized via ZSync protocol."""
 
-    PortType = ZhPort
-
-    def __init__(self, name, device_setup, time_of_flight=0.0, smearing=0.0):
+    def __init__(
+        self,
+        name,
+        device_setup,
+        channels: list[ZiChannel],
+        time_of_flight=0.0,
+        smearing=0.0,
```

Comment:
I guess these are no longer needed here, since we are passing them through the acquisition config.

Reply:
That's true